Projects/UOContent/Multis/Houses/BaseHouse.cs

```diff
@@ -602,12 +604,18 @@ public virtual bool CheckDecay()
 
         public virtual void KillVendors()
         {
-            foreach (var vendor in PlayerVendors.ToList())
+            using var list = PooledRefList<Mobile>.Create();
+            list.AddRange(PlayerVendors);
+
+            foreach (var vendor in list)
             {
-                vendor.Destroy(true);
+                ((PlayerVendor)vendor).Destroy(true);
```

Comment:
I opted to use Mobile here cause then we can reuse existing buckets instead of creating a one-off PlayerVendor bucket cache just for this.